docs/adr/0011-operators.md

```diff
@@ -0,0 +1,145 @@
+# Managing operators
+
+- Status: accepted
+- Deciders: tumido, HumairAK, 4n4nd, anishasthana
+- Date: 2021-01-27
+
+Technical Story: [issue-1](https://github.com/operate-first/apps/issues/126), [pr-1](https://github.com/operate-first/apps/pull/131), [pr-2](https://github.com/operate-first/apps/pull/153)
+
+## Context and Problem Statement
+
+A `cluster-scope` application definition introduced by [ADR 0010][] is a bit vague for use cases like deploying operators. Operator Lifecycle Manager (OLM) defines a resource overlap where a single resource kind can be used in different contexts which may go against the distinction introduced in [ADR 0010][]. This document should provide guidance to how each of the resource kinds for OLM should be treated in each context and usage variant we aim to support.
+
+## Decision Drivers
+
+- The definition of a `cluster-scope` application at [ADR 0010][] does not provide enough guidance for OLM.
+- Various OLM resources can be used in a way which effects the whole cluster or in a way that limits the effects to a single namespace.
+- Various OLM resources have permission restrictions or behave as a singletons within a namespace.
+- Some operators are not deployed via OLM, yet they behave as operators across all namespaces. Placement of those manifests is also disputed.
+- How should we handle meta-operators which operate across all namespaces and deploy other operators that operate across all namespaces?
+
+## Considered Options
+
+1. Deploy all operator-related resources within `cluster-scope` application.
+2. Require all namespaced resources to be deployed within individual applications, outside of `cluster-scope`.
+3. Find balance between 1. and 2. Deploy resources which have effect on a cluster level within `cluster-scope` app (also the namespace-scoped ones). The rest is deployed in individual applications.
+
+## Decision Outcome
+
+Chosen option: _"3. Find balance between 1. and 2."_, because it is desired to keep track of all resources which may effect other users and other namespaces i.e. operators provisioned to manage all cluster namespaces.
+
+Proposed decision defines strict rules for given resources. These rules are subject to future updates if needed, but the repositories and applications are required to consistently enforce them:
+
+### Operator Lifecycle Manager compatible operators
+
+Operator Lifecycle Manager defines following resource kinds:
```

Comment:
Can we link [here](https://docs.openshift.com/container-platform/4.6/operators/understanding/olm/olm-understanding-olm.html) for those that may need a quick review/overview of these resources, that link gives a good summary of all of these.

Reply:
I have that link down in the `Links` section, but I can link it here as well :+1: